src/get_env.c

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "c_utilities/get_env.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef WIN32
+# define WINDOWS_ENV_BUFFER_SIZE 2048
+static char __env_buffer[WINDOWS_ENV_BUFFER_SIZE];
+#endif  // WIN32
+
+
+const char *
+utilities_get_env(const char * env_name, const char ** env_value)
+{
+  if (!env_name) {
+    return "argument env_name is null";
+  }
+  if (!env_value) {
+    return "argument env_value is null";
+  }
+  *env_value = NULL;
+#ifdef WIN32
+  size_t required_size;
+  errno_t ret = getenv_s(&required_size, __env_buffer, sizeof(__env_buffer), env_name);
```

Comment:
Based on the fact that this test passed: https://github.com/ros2/c_utilities/blob/8255f35c36f5ab7c73005839e05b3f319c49867a/test/test_get_env.cpp#L37-L39

I saw that in some places we use [dupenv_s](https://msdn.microsoft.com/en-us/library/ms175774.aspx). Is it for thread safety of just to not have to handle memory  allocation ? I don't mind switching `getenv_s` to `dupenv_s` if there is a strong preference